product_customer_code/product_customer_code.py

```diff
@@ -23,44 +23,44 @@
 
 
-class ProductCustomerCode(osv.Model):
+class ProductCustomerCode(models.Model):
     _name = "product.customer.code"
     _description = "Add manies Code of Customer's"
 
     _rec_name = 'product_code'
 
-    _columns = {
-        'product_code': fields.char('Customer Product Code', size=64,
-                                    required=True,
-                                    help="""This customer's product code
-                                            will be used when searching into
-                                            a request for quotation."""),
-        'product_name': fields.char('Customer Product Name', size=128,
-                                    help="""This customer's product name will
-                                            be used when searching into a
-                                            request for quotation."""),
-        'product_id': fields.many2one('product.product', 'Product',
-                                      required=True),
-        'partner_id': fields.many2one('res.partner', 'Customer',
-                                      required=True),
-        'company_id': fields.many2one('res.company', 'Company',
-                                      required=False),
-    }
+    product_code = fields.Char('Customer Product Code', size=64, required=True,
+                               help="""This customer's product code will be
+                               used when searching into a request for
+                               quotation.""")
 
-    _defaults = {
-        'company_id': lambda s, cr,
-        uid, c: s.pool.get('res.company').
-        _company_default_get(cr, uid,
-                             'product.customer.code',
-                             context=c),
-    }
+    product_name = fields.Char('Customer Product Name', size=128,
+                               help="""This customer's product name will
+                               be used when searching into a request for
+                               quotation.""")
+
+    product_id = fields.Many2one('product.product', 'Product', required=True,
+                                 help='Product Identification')
+
+    partner_id = fields.Many2one('res.partner', 'Customer', required=True,
+                                 help='Partner Reference')
+
+    @api.model
+    def default_get(self, default_fields):
+        rec = super(ProductCustomerCode, self).default_get(default_fields)
+        rec['company_id'] = (
+            self.env['res.company']._company_default_get(
+                'product.customer.code'))
+        return rec
+
+    company_id = fields.Many2one('res.company', 'Company',
+                                 help='Company Reference',
+                                 )
 
     _sql_constraints = [
         ('unique_code', 'unique(product_code,company_id,partner_id)',
          'Product Code of customer must be unique'),
     ]
-
-    # TODO: Add index to product_code, partner_id
```

Comment:
You are removing the `TODO` but I didn't see the `index` parameter for these fields